### ISSUE TITLE
Setting the text of badge, adds another instance of span with the new value

### ISSUE THREAD

**Issue by HerbertsVaadin** [TatuLund/badge#3]
In version 2.2.2 (V23 branch)
```java
    /**
     * Set the text content of the badge, remove all other content.
     * 
     * @param text
     *            String value.
     */
    public void setText(String text) {
        this.text = text;
        add(new Span(text));
    }
```
Doesn't remove old content as stated in javadoc,
so we just get more spans and texts added every time we call `badge.setText(..)`.

![image](https://github.com/TatuLund/badge/assets/80950643/0806536e-0a0e-47ed-aca2-6f1b5eafd1b1)

**Comment by TatuLund** [comment]
Good catch.

**Comment by TatuLund** [comment]
Fixed in 2.2.3 / 2.30
